opensearch_py_ml/ml_commons/rag_pipeline/rag/ingest.py

```diff
@@ -0,0 +1,241 @@
+# SPDX-License-Identifier: Apache-2.0
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+# Any modifications Copyright OpenSearch Contributors. See
+# GitHub history for details.
+
+
+#  Licensed to Elasticsearch B.V. under one or more contributor
+#  license agreements. See the NOTICE file distributed with
+#  this work for additional information regarding copyright
+#  ownership. Elasticsearch B.V. licenses this file to you under
+#  the Apache License, Version 2.0 (the "License"); you may
+#  not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+# 	http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing,
+#  software distributed under the License is distributed on an
+#  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+#  KIND, either express or implied.  See the License for the
+#  specific language governing permissions and limitations
+#  under the License.
+
+import os
+import glob
+import json
+import tiktoken
+from tqdm import tqdm
+from colorama import Fore, Style, init
+from typing import List, Dict
+import csv
+import PyPDF2
+import boto3
+import botocore
+import time
+import random
+
+
+from opensearch_connector import OpenSearchConnector
+
+init(autoreset=True)  # Initialize colorama
```

Comment:
what is this for?

Reply:
This shows when the documents are ingested. If the documents are successfully ingested, it shows green; if they fail, it shows red.